core/src/tasks/profileProperty/importProfileProperty.ts

```diff
@@ -46,7 +47,19 @@ export class ImportProfileProperty extends RetryableTask {
       if (properties[dep.key].state !== "ready") ok = false;
     });
 
-    if (!ok) return; // there's a dependency we don't have yet
+    if (!ok) {
+      // there's a dependency we don't have yet, sleep a little and will be retried later
+      return ProfileProperty.update(
+        { startedAt: ImportOps.retryStartedAt() },
```

Comment:
I'd be ok merging too, but noting this is technical debt and a weird spot.

Reply:
Yeah - it felt weird.  I like option 1, keeping the 5 minute retry as a fallback if redis goes away

Reply:
The trick would be to only do that a few times (so there isn't one in resque while it's getting re-enqueued) because it's been five minutes.
So that might look like checking when that would be (by getting the setting and adding it to startedAt). if that's "soon" (within 20 seconds), drop it on the floor. If it's more than that. re-enqeue for 10 seconds later. You could also check the length of how many are pending to get a sense.
The main downside of this approach is that this this re-enqueued batch would be smaller and you'd end up with lots of little ones. Whereas the messing with the db ones (including the startedAt approach) keeps big batch sizes.